packages/evm/contracts/core/modules/ACL.sol

```diff
@@ -0,0 +1,96 @@
+pragma solidity 0.5.17;
+
+
+contract ACL {
+    string private constant ERROR_BAD_FREEZE = "ACL_BAD_FREEZE";
+    string private constant ERROR_ROLE_GRANTED = "ACL_ROLE_GRANTED";
+    string private constant ERROR_ROLE_NOT_GRANTED = "ACL_ROLE_NOT_GRANTED";
+    string private constant ERROR_ROLE_ALREADY_FROZEN = "ACL_ROLE_ALREADY_FROZEN";
+    string private constant ERROR_INVALID_BULK_INPUT = "ACL_INVALID_BULK_INPUT";
+
+    enum BulkOp { Grant, Revoke, Freeze }
+
+    address internal constant FREEZE_FLAG = address(1);
+    address internal constant ANY_ADDR = address(-1);
+
+    // List of all roles assigned to different addresses
+    mapping (bytes32 => mapping (address => bool)) public roles;
+
+    event Granted(bytes32 indexed id, address indexed who);
+    event Revoked(bytes32 indexed id, address indexed who);
+    event Frozen(bytes32 indexed id);
+
+    modifier notFrozen(bytes32 _id) {
+        require(!isRoleFrozen(_id), ERROR_ROLE_ALREADY_FROZEN);
```

Comment:
Curious if you like this modifier more than just inlining it.

If we do, let's also make the change in Govern's ACL so it's checked in a similar manner.